Rebus.AdoNet/AdoNetTimeoutStorage.cs

```diff
@@ -80,50 +82,99 @@ public void Add(Timeout.Timeout newTimeout)
 			}
 		}
 
+		public string GetSqlForDialect(Version version)
+		{
+			return dialect.GetSql(version);
+		}
+
 		/// <summary>
 		/// Queries the underlying table and returns due timeouts, removing them at the same time
 		/// </summary>
 		public DueTimeoutsResult GetDueTimeouts()
 		{
 			var dueTimeouts = new List<DueTimeout>();
+			IDbConnection connection = null;
+			IDbTransaction transaction = null;
 
-			using (var connection = factory.OpenConnection())
-			using (var command = connection.CreateCommand())
-			{
-				const string sql = @"
-SELECT ""id"", ""time_to_return"", ""correlation_id"", ""saga_id"", ""reply_to"", ""custom_data""
-FROM ""{0}""
-WHERE ""time_to_return"" <= @current_time
-ORDER BY ""time_to_return"" ASC
-";
-
-				command.CommandText = string.Format(sql, timeoutsTableName);
-
-				command.AddParameter("current_time", RebusTimeMachine.Now());
-
-				using (var reader = command.ExecuteReader())
-				{
-					while (reader.Read())
-					{
-						var sqlTimeout = DueAdoNetTimeout.Create(MarkAsProcessed, timeoutsTableName, reader);
+			try
+			{
+				connection = factory.OpenConnection();
+				transaction = connection.BeginTransaction();
+
+				using (var command = connection.CreateCommand())
+				{
+					command.Transaction = transaction;
 
-						dueTimeouts.Add(sqlTimeout);
-					}
+					var version = new Version(dialect.GetDatabaseVersion(connection));
```

Comment:
1) This class should not depend on postgresql version. That's what Dialects are supposed to abstract.
2) We should not be asking for version on each iteration.. this is suboptimal.